sensor/common/component.go

```diff
@@ -10,7 +10,7 @@ import (
 // as well as sending messages back to central.
 type SensorComponent interface {
 	Start() error
-	Stop(err error)
+	Stop(err error) // TODO: get rid of err argument as it always seems to be effectively nil.
```

Comment:
I bet we forget about this TODO. Could you maybe create a ticket for it and link it? Or at least start a new draft PR for this?

Reply:
+1